hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java

```diff
@@ -1,73 +1,82 @@
+
+		// Setup web endpoint config
+		setupSSLContextHibcRegistry(getContext());
+		String hibcHttpUrl = String.format(
+				properties.getValue(HIBC_HTTP_URI) + "?bridgeEndpoint=true&sslContextParameters=#%s&authMethod=Basic&authUsername=%s&authPassword=%s",
+				SSL_CONTEXT_HIBC, properties.getValue(HIBC_USER), properties.getValue(HIBC_PASSWORD));
+		String basicAuthToken = WebAuthenticationUtils.buildBasicAuthToken(properties.getValue(HIBC_USER), properties.getValue(HIBC_PASSWORD));
+
+		// Setup MQ config
 		String hibcRequestQueue = properties.getValue(HIBC_REQUEST_QUEUE);
 		String hibcReplyQueue = properties.getValue(HIBC_REPLY_QUEUE);
 		String hibcMqUrl = String.format(MQ_URL_FORMAT, hibcRequestQueue, hibcReplyQueue);
 
 		handleExceptions();
 
 		from("direct:hibc").routeId("hibc-route")
-			.process(new ProtocolEvaluator()).id("ProtocolEvaluator")
+			.process(new ProtocolEvaluator()).id("ProtocolEvaluator") // Check if message type is configured for MQ or HTTP
 			.choice()
 				.when(exchangeProperty(Util.PROPERTY_MESSAGE_PROTOCOL).isEqualTo(Util.PROTOCOL_HTTP))
-					.to("direct:hibcHTTP")
+					.to(DIRECT_HIBC_HTTP)
 				.when(exchangeProperty(Util.PROPERTY_MESSAGE_PROTOCOL).isEqualTo(Util.PROTOCOL_MQ))
 					.to(DIRECT_HIBC_MQ)
 				.otherwise()
-					.log("Protocol for HIBC message type ${exchangeProperty.messageType} not found or not valid. Defaulting to MQ")
-					.to(DIRECT_HIBC_MQ)
+					.log("Protocol for HIBC message type ${exchangeProperty.messageType} not found or not valid. Defaulting to HTTP")
+					.to(DIRECT_HIBC_HTTP)
 			.end();
 
-		// XXX This is currently just a simple HTTP route which will need additional configuration (auth, request/response conversion)
-		// once the endpoint is available
-		from("direct:hibcHTTP").routeId("hibc-http-route")
-	     	.to("log:HttpLogger?level=DEBUG&showBody=true&multiline=true")           		
-	     	.log("Sending to HIBC")
+		from(DIRECT_HIBC_HTTP).routeId("hibc-http-route")
+	     	.to("log:HttpLogger?level=DEBUG&showBody=true&multiline=true")
 	     	.process(new AuditSetupProcessor(TransactionEventType.MESSAGE_SENT))
 	     	.wireTap(DIRECT_AUDIT).end()
-	     	.setBody().method(new Base64Encoder()).id("HIBCBase64Encoder")
-            .setBody().method(new ProcessV2ToJson()).id("HIBCProcessV2ToJson")
 	     	.setHeader(Exchange.CONTENT_TYPE, constant("application/json"))
-	        .setHeader("CamelHttpMethod", POST)
-	     	.to("log:HttpLogger?level=INFO&showBody=true&showHeaders=true&multiline=true")
+	        .setHeader(CAMEL_HTTP_METHOD, POST)
+			.setHeader(AUTHORIZATION, simple(basicAuthToken))
+	     	.to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
 	     	.to(hibcHttpUrl).id("ToHibcHttpUrl")
-	     	.log("Received response from HIBC")
-	     	.bean(FhirPayloadExtractor.class)
-	     	.log("Decoded V2: ${body}")
+			.to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
```

Comment:
Is one of these two log statements redundant? Edit: nevermind, I guess one is for the request and one for the response.